script/validate

```diff
@@ -9,19 +9,24 @@ import yaml
 from jsonschema import validate
 
 
-def extract_urns(data, urns=None):
+def extract_urns(data, urns=None, seen_urns=None):
     if urns is None:
         urns = []
+    if seen_urns is None:
+        seen_urns = set()
 
     if isinstance(data, dict):
         for key, value in data.items():
             if key == "urn":
-                urns.append(value)
-            else:
-                extract_urns(value, urns)
+                if value in seen_urns:
+                    urns.append(value)
+                else:
+                    seen_urns.add(value)
```

Comment:
I'm not sure I understand this logic. Does this require the urn to be present twice before it gets added to `urn`?

Reply:
To ensure we can reference an urn from another task within the same instrument, I've added the this logic. 

This is needed to support the conformity assessment; we need to refer to different tasks that the are 'signed' by another task within the same instrument. However, we could consider other methods of referencing then urns within a instrument, if more suitable and robust.